src/goto-programs/show_properties.cpp

```diff
@@ -23,6 +23,30 @@ Author: Daniel Kroening, kroening@kroening.com
 #include "goto_functions.h"
 #include "goto_model.h"
 
+
+source_locationt find_property(
```

Comment:
Suggest `optionalt<source_locationt>` would be better than throwing a C-string

Reply:
I can't think of a situation where you would want to continue if the property that you are looking for doesn't exist? So if I make this an optionalt<source_locationt>, I will just end up throwing an exception wherever the return value is used.

Actually, that leads onto a general comment on goto-instrument behaviour. If I do a reachability-slice and specify a given property, and that property is not found, I would expect some kind of error message saying "property not found". What actually happens is it that it happily does the reachability slice for all properties in the binary, without telling you that is what it has done.

Reply:
Yeah I think the caller should throw, this should expose an interface friendlier to people searching for a property that may or may not exist in a particular program

Reply:
ok, changed